src/core/lib/pick-layers.js

```diff
@@ -123,7 +123,7 @@ export function pickObject(gl, {
       pickedLayer === layer &&
       pickingSelectedColor !== EMPTY_PIXEL
     );
-    // TODO - handle multi model layers?
+    // Note: Auto highlighting only works for single model layers
```

Comment:
Why does it only work for single model layers?

Reply:
> Why does it only work for single model layers?

In general it is not clear to me that our picking system works at all for multi model layers. I believe we assign the alpha picking identifier per layer, not per model, so there is no way to distinguish which model was picked, and AFAIK `pickInfo` contains no provisions for communicating it either.

Reply:
True there may be more work required to make picking work in a multi-model layer, but it's definitely doable.
- A layer owns the picking color encoding/decoding process, therefore it could use the encoding to differentiate which model is picked.
- From the user's perspective, `pickInfo` only need to contain an object from the `data` array, and should never expose implementation details such as source model.

In practice, most multi-model layers I've written use multiple primitive models to render a more complex mesh. Models in this case share most attributes such as `instancePosition` and `instancePickingColor`, and picking already works out of the box. The code in question here makes it impossible for these layers to leverage in the auto highlight system.

Reply:
+1, picking/highlighting system should work with multi models , as long as the layer encodes/decodes `instancingPickingColor` differently for each instance of the model.

> The code in question here makes it impossible for these layers to leverage the auto highlight system.

All this code does, checks layer's props and returns moduleSetting needed for auto-highlighing, this method doesn't have to access layer.state.mode, we can move this into layer class (as a public method, takes `drawPickingColors`  and return the moduelSettings, which can be merged with with `moduleParameters` in above 'drawLayersInViewport' method. So this code doesn't have to access `layer.state.model`.

Reply:
> as long as the layer encodes/decodes instancingPickingColor differently for each instance of the model.

I believe that this is not what this PR does. The assumption here is that in a multi model layer all `Model`s have the same sets of "indices", corresponding to the same objects in the same data array, and should be highlighted at the same time. (i.e. the different models do not show different parts of the data array).

I think this is a valid choice, and I do not have strong opinions either way, but it should probably be mentioned somewhere in the docs.